### PR TITLE
Nix.Builtins: add getContext and appendContext primops

### PR DESCRIPTION
**Pull request by vaibhavsagar**

### CODE REVIEW

**Review by jwiegley — CHANGES_REQUESTED** [review]
Can you please add tests for these two new builtins?

**Review by jwiegley — DISMISSED** [review]

**Review by jwiegley — DISMISSED** [review]

File: tests/eval-compare/builtins.appendContext.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+let
```
Comment (shlevy) [inline tests/eval-compare/builtins.appendContext.nix:1]: Why do we need this test when we have the upstream one?

Comment (mightybyte) [reply tests/eval-compare/builtins.appendContext.nix:1]: Oh, good point.  @jwiegley requested tests and I guess I  momentarily forgot that we have NixLanguageTests.

**Comment by jwiegley** [comment]
@vaibhavsagar I've merged this by hand, and updated it to match some recent changes to the code. However, I've had to comment out the main bits of the `appendContext` test, since it's not matching the output from `nix-instantiate`. Can you take a look?

**Comment by vaibhavsagar** [comment]
cc @kmicklas

**Pull request merged**